File: 0860-lemonade-change/0860-lemonade-change.py
class Solution:
    def lemonadeChange(self, bills: List[int]) -> bool:
        count = [0] * 21
        for bill in bills:
            count[bill] += 1
            if bill > 10 and count[10] > 0:
                bill -= 10
                count[10] -= 1
            while bill > 5 and count[5] > 0:
                bill -= 5
                count[5] -= 1
            if bill > 5: return False
        return True

File: 0860-lemonade-change/0860-lemonade-change.java
class Solution {
    public boolean lemonadeChange(int[] bills) {
        var count = new int[21];
        for(var bill : bills){
            count[bill]++;
            if(bill > 10 && count[10] > 0){
                bill -= 10;
                count[10]--;
            }
            while(bill > 5 && count[5] > 0){
                bill -= 5;
                count[5]--;
            }
            if(bill > 5) return false;
        }
        return true;
    }
}